diffmpm/particle.py

```diff
@@ -1,15 +1,66 @@
 import jax.numpy as jnp
 
-class Particle1D:
-    def __init__(self, mass, x, material):
-        self.id = None
-        
+
+class Particles:
+    """
+    Container class for particles on a mesh.
+    """
+
+    def __init__(
+        self,
+        mass,
+        x,
+        xi,
+        material,
+        ppe,
+        nelements,
+        element_ids,
+        domain_size,
+        ptype="uniform",
+    ):
+        """
+        Construct a container for particles.
+
+        Arguments
+        ---------
+        mass : float, array_like
+            Mass of each particle. Can be a float or an array for mass of each particle.
+        x : array_like
+            Position of particles in physical coordinates.
+        xi : array_like
+            Position of particles in natural coordinates.
+        material : diffmpm.material.Material
+            Material type of the mesh the particles are a part of.
+        ppe : int
+            Number of particles per element.
+        nelements : int
+            Number of elements that contain the particles.
+        element_ids : array_like
+            Ids of the elements that each particle is a part of.
+        domain_size : float
+            Size of the domain in consideration.
+        """
+        # self.mass = (
+        #     jnp.empty(ppe * nelements).fill(mass)
+        #     if jnp.isscalar(mass)
+        #     else jnp.asarray(mass)
+        # )
+        self.material = material
+        self.nparticles = ppe * nelements
+        self.x = x
+        self.xi = xi
+        self.element_ids = element_ids
         self.mass = mass
-        self.x  = x 
-        self.material = material 
-        self.density = material.density 
```

Comment:
Why did you remove density, we need to compute mass of particle

Reply:
I compute the mass of the particle while initializing and store it. Also, since it already has material, looking up density would be better to look up the density from the material object instead of storing it separately? My thought was that this way the density cannot be changed without affecting the material.